test_01_optimal.py

```diff
@@ -18,10 +18,10 @@ def test_and():
     assert gate_count(And) == 2
 
 def test_xor():
-    assert gate_count(Xor) == 5
+    assert gate_count(Xor) <= 5
 
 def test_mux():
-    assert gate_count(Mux) == -1 # < 8
+    assert gate_count(Mux) <= 8
```

Comment:
This has been updated to `== 4` on the `solved` branch, but I missed propagating that back to `master`. Doing this with branches is turning out to be a drag. Maybe I should just put the solutions in `master`, but hide them in another file? I wanted to give people a chance to not peek.